### DISCUSSION TITLE
webgl context 复用

### DISCUSSION THREAD

**Discussion by BUPTlhuanyu** [galacean/effects-runtime#720] [Ideas]
https://github.com/galacean/effects-runtime/blob/6734dcb03250caf597f246d442fad2a7378788c0/packages/effects-webgl/src/gl-renderer.ts#L47C24-L47C40

由于浏览器对 webgl 数量有限制，这里是否可以加上 webgl 缓存池用于 webgl 的复用？

**Comment by yiiqii** [comment]
不可以哦，一条渲染管线的建立到结束是和 context 直接绑定的

**Reply by BUPTlhuanyu** [reply]
如果是对同一个渲染资源也不能复用么？需要业务手动复用是吧？

**Reply by yiiqii** [reply]
是的哦

**Reply by BUPTlhuanyu** [reply]
实际业务中有存在多个动效弹窗的情况吗？这块各个机型有测试过吗？

**Reply by yiiqii** [reply]
一般一次弹一个窗（即：下面一个 canvas，上面一个 canvas），这种场景有大体量设备验证

**Reply by BUPTlhuanyu** [reply]
依次弹不同物料的弹窗也是有类似验证是吗？

**Reply by yiiqii** [reply]
依次弹但是会移除上一个弹窗吧？同时不会有多个弹窗共存的

**Reply by BUPTlhuanyu** [reply]
会移除弹窗，但是 webgl context 在移动端 safrai 上看还存留的，我这是简单测试了一下，不知道真正业务里用的时候，弹出关闭很多次弹窗会不会有啥问题

**Reply by yiiqii** [reply]
安卓上有销毁不及时的问题，iOS 没有遇到，建议每次弹窗复用同一个 player 示例，这样无论多少个弹窗也不会有问题

**Reply by BUPTlhuanyu** [reply]
好的，感谢

**Reply by BUPTlhuanyu** [reply]
我又来了。。。
目前你们用到的场景在 ios 上的 safrai 上图形标签 webgl context 会被清除吗？求个使用了这个渲染引擎的线上项目地址？

**Reply by yiiqii** [reply]
都是在移动端使用的，淘宝（iOS、安卓）、支付宝等 App 的双十一首页或相关子页面都是

**Reply by BUPTlhuanyu** [reply]
好的，多谢，我去找找
